pkg/core/operation_test.go

```diff
@@ -97,6 +98,45 @@ func TestOperationTypes(t *testing.T) {
 	assert.False(t, op.IsBlockchainOperation())
 }
 
+func TestOperationDeepCopy(t *testing.T) {
+	op := &Operation{
+		ID:          fftypes.NewUUID(),
+		Namespace:   "ns1",
+		Transaction: fftypes.NewUUID(),
+		Type:        OpTypeBlockchainInvoke,
+		Status:      OpStatusInitialized,
+		Plugin:      "fake",
+		Input:       fftypes.JSONObject{"key": "value"},
+		Output:      fftypes.JSONObject{"result": "success"},
+		Error:       "error message",
+		Created:     fftypes.Now(),
+		Updated:     fftypes.Now(),
+		Retry:       fftypes.NewUUID(),
+	}
+
+	copyOp := op.DeepCopy()
+
+	assert.Equal(t, op.ID, copyOp.ID)
+	assert.Equal(t, op.Namespace, copyOp.Namespace)
+	assert.Equal(t, op.Transaction, copyOp.Transaction)
+	assert.Equal(t, op.Type, copyOp.Type)
+	assert.Equal(t, op.Status, copyOp.Status)
+	assert.Equal(t, op.Plugin, copyOp.Plugin)
+	assert.Equal(t, op.Input, copyOp.Input)
+	assert.Equal(t, op.Output, copyOp.Output)
+	assert.Equal(t, op.Error, copyOp.Error)
+	assert.Equal(t, op.Created, copyOp.Created)
+	assert.Equal(t, op.Updated, copyOp.Updated)
+	assert.Equal(t, op.Retry, copyOp.Retry)
+
+	// Ensure the copy is a deep copy
+	op.ID = fftypes.NewUUID()
+	assert.NotEqual(t, copyOp.ID, op.ID)
```

Comment:
This is not really proving what the function should do. You should really not reassign the field but instead compare the address of the returning fields `assert.NotEqual(t, &op.ID, &copyOp.ID)`

Your test is working because you copying the whole struct but not modifying the data of a field in place, because the `fftypes.UUID` and `fftypes.FFTime` do not have a mutating function we are "covered"